### ISSUE TITLE
[Bug Report]: My ipega controller does not connect with switch in atmosphere

### ISSUE THREAD

**Issue by Kayorock** [ndeadly/MissionControl#602]
### Switch Firmware Version

16.0.0 (Latest)

### Atmosphère Version

1.5.0 (Latest)

### Mission Control Version

0.9.0 (Latest)

### Boot Method

Hekate

### Issue Description

When I try to connect my ipega 9076 control it doesn't work, it is not the standard one, it has an icon with LT OMEX written on it (which apparently is from a distributor called altomex) underneath the name ipega, I tried to put it in bluetooth mode and also by the USB adapter but the switch doesn't recognize it and doesn't connect either way.

**Comment by ndeadly** [comment]
USB adapter? If this controller comes with its own USB dongle, then it's quite possible it doesn't support Bluetooth at all. Can you connect it to a PC without USB? All compatible Bluetooth controllers should pair with the Switch, regardless of whether Mission Control officially supports them. Unsupported controllers will connect, but their buttons won't be mapped to anything, so they won't respond to inputs.

**Comment by Kayorock** [comment]
It has the USB dongle but it also works via bluetooth on my cell phone and on the pc too, it's the ipega 9076 so different from the common one this one comes with an altomex logo, but on the switch it doesn't work either on the dock with dongle or via bluetooth

**Comment by Kayorock** [comment]
here some pictures of him.

![IMG_20230329_152120489](https://user-images.githubusercontent.com/129319305/228632296-1f88f227-6db3-49f2-9a82-89ee7d334441.jpg)
![IMG_20230329_152134849](https://user-images.githubusercontent.com/129319305/228632309-9337f29d-710f-4192-95de-888bba0a50e4.jpg)
![IMG_20230329_152144524](https://user-images.githubusercontent.com/129319305/228632314-b74b5695-f1d8-4b75-b5c7-5c1e00f0ad0b.jpg)

**Comment by ndeadly** [comment]
> but on the switch it doesn't work either on the dock with dongle or via bluetooth

The dongle won't do anything with Mission Control as it's effectively a USB controller with a wireless link. Via bluetooth it should at least connect to the console if you're doing everything correctly. Have you followed the instructions for [pairing controllers](https://github.com/ndeadly/MissionControl#pairing-controllers) in the readme? You should be connecting the controller in Android mode as far as I'm aware.

**Comment by Kayorock** [comment]
Yes, I tried to stop in bluetooth mode too but unfortunately it doesn't connect, when I was on firmware 15.0.1 it didn't work either, I updated it to firmware 16.0.0 and it still doesn't work.

**Comment by Kayorock** [comment]
One detail that I noticed when I connected it to my cell phone is that its name here is pg 9156 and not 9076, I don't know if that interferes with anything.

**Comment by impeeza** [comment]
Maybe the controller is using BLE?

**Comment by ndeadly** [comment]
Yes, I wouldn't expect this to be any different on PC, it's simply the device name it's advertising. What I'm interested in is whether it's an LE controller or not. If you're on windows you should be able to check this by finding the device under Device Manager->Bluetooth and going to Properties->Details->Device description or similar and looking for mentions of Bluetooth LE.

Regarding the firmware, you could try updating if you like, but I really don't know what's going on with the versioning here. It could be safer to leave it since it seems a bit unclear what device it actually is.

**Comment by Kayorock** [comment]
I did the step by step and I got there.

![IMG_20230330_180203606](https://user-images.githubusercontent.com/129319305/228963751-b1800595-91b7-4032-9056-27fb4d582d31.jpg)

**Comment by ndeadly** [comment]
Ah there we go, it's using Bluetooth LE, @impeeza was right. You'll have to wait until LE support has been added.

**Comment by Kayorock** [comment]
OK, I'll be looking forward to playing with this controller on my Switch :)
